kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/Coordinator.java

```diff
@@ -218,6 +220,10 @@ private void commitToTable(
             .filter(distinctByKey(deleteFile -> deleteFile.path().toString()))
             .collect(Collectors.toList());
 
+    if (terminated) {
+      throw new ConnectException("Coordinator is terminated, commit aborted");
```

Comment:
nit: Do you think adding a debug log would be helpful here?

Reply:
I don't feel it is necessary to both throw an exception and have a log statement, the exception should ultimately get logged